application_boilerplate/js/MapUrlParams.js

```diff
@@ -0,0 +1,290 @@
+              wkid = markerArray[2] || null,
+              description = markerArray[3] || null,
+              icon_url = markerArray[4] || this.defaultMarkerSymbol || null,
+              label = markerArray[5] || null;
+
+
+              var markerSymbol = new PictureMarkerSymbol(icon_url, 26, 26);
+              var point = new Point({
+                  "x": x,
+                  "y": y,
+                  "spatialReference": {
+                      "wkid": wkid || 4326
+                  }
+              });
+
+              var infoTemplate = null;
+              if (description || label) {
+                  infoTemplate = new PopupTemplate({
+                      "title": label || null,
+                      "description": description || null
+                  });
+              }
+              this._project(point).then(lang.hitch(this, function(pPoint){
+                var projectedGraphic = new Graphic(pPoint, markerSymbol, null, infoTemplate);
+                var projectedResult = (pPoint) ? {"markerGraphic": projectedGraphic} : null;
+                deferred.resolve(projectedResult);
+              }),deferred.reject);
+         }else{
+           deferred.reject();
+         }
+      }));
+      return deferred.promise;
+    },
+    _splitArray: function(value){
+      var splitValues = value.split(";");
+      if(splitValues.length === 1){
+        splitValues = value.split(",");
+      }
+      return splitValues;
+    },
+    _project: function(geometry){
+      var deferred = new Deferred();
+      if(this._sameSpatialReference(geometry.spatialReference, this.mapSpatialReference)){
+          deferred.resolve(geometry);
+      }else{
+        esriConfig.defaults.geometryService.project([geometry], this.mapSpatialReference).then(lang.hitch(this, function(projectedGeometry){
+            if(projectedGeometry && projectedGeometry.length && projectedGeometry.length > 0){
+              deferred.resolve(projectedGeometry[0]);
+            }else{
+              deferred.reject();
+            }
+        }),function(error){
+          deferred.reject(error);
+        });
+      }
+      return deferred.promise;
+    },
+    _sameSpatialReference: function(sp1, sp2) {
+      var same = false;
+      var mercator = [102113, 102100, 3857];
+      if (sp1 && sp2 && sp1.wkt == sp2.wkt && (sp1.wkid == sp2.wkid || (esriLang.isDefined(sp1.latestWkid) && sp1.latestWkid == sp2.wkid) || (esriLang.isDefined(sp2.latestWkid) && sp1.wkid == sp2.latestWkid) || (esriLang.isDefined(sp1.latestWkid) && sp1.latestWkid == sp2.latestWkid))) {
+        same = true;
+      } else if (sp1 && sp2 && sp1.wkid && sp2.wkid && this._contains(mercator, sp1.wkid) && this._contains(mercator, sp2.wkid)) {
+        same = true;
+      }
+      return same;
+    },
+    _contains: function(array, obj){
+      var cont = false;
+      var i = array.length;
+      while (i--){
+        if(array[i] === obj){
+          cont = true;
```

Comment:
Probably I'll take a look. I was just being lazy and copied the code verbatim from online. :)